### PR TITLE
Migrate to babel-preset-env.

### PR DESCRIPTION
**Pull request by rwjblue**
Prior to this PR:

```
> ember build --environment=production
cleaning up...
Built project successfully. Stored in "dist/".
File sizes:
 - dist/app-1a294d2bf122a6c7c5987eb976e42e67.css: 22 B (42 B gzipped)
 - dist/app-35053405b2117ac81ef9572e6ed78c38.js: 131.66 KB (32 KB gzipped)
```

After the PR (with required update to uglify-es to support minifying ES6 code):

```
> ember build --environment=production
cleaning up...
Built project successfully. Stored in "dist/".
File sizes:
 - dist/app-1a294d2bf122a6c7c5987eb976e42e67.css: 22 B (42 B gzipped)
 - dist/app-bb2c72327987b1adaed6c6934e594a42.js: 108 KB (29.62 KB gzipped)
```

**tldr;** Roughly 8% savings overall...

### CODE REVIEW

**Comment by rwjblue** [comment]
Prior to this PR:

```
> ember build --environment=production
cleaning up...
Built project successfully. Stored in "dist/".
File sizes:
 - dist/app-1a294d2bf122a6c7c5987eb976e42e67.css: 22 B (42 B gzipped)
 - dist/app-35053405b2117ac81ef9572e6ed78c38.js: 131.66 KB (32 KB gzipped)
```

After the PR (with required update to uglify-es to support minifying ES6 code):

```
> ember build --environment=production
cleaning up...
Built project successfully. Stored in "dist/".
File sizes:
 - dist/app-1a294d2bf122a6c7c5987eb976e42e67.css: 22 B (42 B gzipped)
 - dist/app-bb2c72327987b1adaed6c6934e594a42.js: 108 KB (29.62 KB gzipped)
```

**tldr;** Roughly 8% savings overall...
